tensorboard/plugins/graph/tf_graph_common/minimap.ts

```diff
@@ -155,7 +155,23 @@ export class Minimap {
     let $download = d3.select('#graphdownload');
     this.download = <HTMLLinkElement>$download.node();
     $download.on('click', d => {
-      this.download.href = this.downloadCanvas.toDataURL('image/png');
+      // Revoke the old URL, if any. Then, generate a new URL.
+      URL.revokeObjectURL(this.download.href);
+      // We can't use the `HTMLCanvasElement.toBlob` API because it does
```

Comment:
FWIW, for the scalar plot download buttons, we use a drop-down menu as the trigger to generate the URL so that it's a two-click process.  If we did something similar for the graph it would be possible to use toBlob() and just disable the actual download click target until the synchronous call completes.

Maybe it's not worth it, but I have to imagine toBlob() might be faster than iterating over each character in the blob by hand.